theseus/core/loss_function.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import List, Optional, Tuple, cast
+
+import torch
+
+from theseus.core.theseus_function import TheseusFunction
+
+from ..geometry import Vector
+
+
+class LossFunction(TheseusFunction, abc.ABC):
+    def __init__(self, name: Optional[str] = None):
+        super().__init__(name=name)
+
+    @abc.abstractmethod
+    def function_value(self, error: torch.Tensor) -> torch.Tensor:
+        pass
+
+    @abc.abstractmethod
+    def rescale(
+        self,
+        jacobians: List[torch.Tensor],
+        error: torch.Tensor,
+    ) -> Tuple[List[torch.Tensor], torch.Tensor]:
+        pass
+
```

Comment:
Add a `_copy_impl` abstract method here like [this](https://github.com/facebookresearch/theseus/blob/main/theseus/core/cost_function.py#L62).

Reply:
Will do.